api/app/jobs/noaa.py

```diff
@@ -177,12 +196,11 @@ def parse_url_for_timestamps(url: str, model_type: ModelEnum):
 def parse_gfs_url_for_timestamps(url: str):
     """ Interpret the model_run_timestamp and prediction_timestamp from a GFS model's URL """
     # sample URL: 'https://nomads.ncep.noaa.gov/cgi-bin/filter_gfs_0p25.pl?dir=%2Fgfs.20230412%2F00%2Fatmos&file=gfs.t00z.pgrb2.0p25.f018var_APCP=on&var_RH=on&var_TMP=on&var_UGRD=on&var_VGRD=on&lev_surface=on&lev_2_m_above_ground=on&lev_10_m_above_ground=on&subregion=&toplat=60&leftlon=-139&rightlon=-114&bottomlat=48'
-    model_run_datetime_str = re.search("dir=\%2Fgfs.\d*\%2F\d*", url).group(0)
-    # extract date string only from gfs.20230411 in sample URL
-    model_run_date = model_run_datetime_str[-13:-5]
-    model_run_hour = model_run_datetime_str[-2:]
-    forecast_hour_str = re.search("file=gfs.t\d*z.pgrb2.0p25.f\d*", url).group(0)
-    forecast_hour = forecast_hour_str[-3:]
+    query = urlsplit(url).query
+    params = parse_qs(query)
+    model_run_date = params['dir'][0][5:13]
+    model_run_hour = params['dir'][0].split('/')[2]
+    forecast_hour = params['file'][0][-3:]
```

Comment:
Can we split the tokens on '.' instead of hard coded offsets? Easier to read and and more resilient to string changes.

Reply:
done